packages/flutter/lib/src/material/carousel.dart

```diff
@@ -205,7 +207,108 @@ class CarouselView extends StatefulWidget {
     this.enableSplash = true,
     required List<int> this.flexWeights,
     required this.children,
-  }) : itemExtent = null;
+  }) : itemExtent = null,
+       itemBuilder = null,
+       itemCount = null;
+
+  /// Creates a scrollable carousel with fixed-sized items created on demand.
+  ///
+  /// This constructor allows lazy loading of carousel items. Only items that
+  /// are visible (or about to be visible) are built, improving performance
+  /// when dealing with large numbers of items.
+  ///
+  /// The [itemBuilder] callback will be called only with indices greater than
+  /// or equal to zero and less than [itemCount].
+  ///
+  /// The [itemBuilder] should return null if asked to build an item with an
+  /// index greater than or equal to [itemCount].
```

Comment:
Elsewhere your docs say that this will never happen though?

Reply:
Yup, forgot to update doc. This was for initial implementation.